src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -49,311 +41,320 @@ DeviceInformation Plugin::ParseBatchDevice(const std::string& deviceWithBatch) {
             // check the batch dim: either 0th (and the original batch size of 1) or none
             if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
                 if (shape[0] != 1)
-                    IE_THROW(NotImplemented) << "Auto-batching does not reshape/re-batch originally batched networks!";
+                    OPENVINO_THROW("Auto-batching does not reshape/re-batch originally batched networks!");
                 const auto& node = output->input_value(0);
                 batched_outputs.insert(
                     ov::op::util::get_ie_output_name(ov::Output<const ov::Node>(node.get_node(), node.get_index())));
             } else {
                 // if the 0-th dim is not for the batch, then we support only the case when NONE dimension is batch
                 for (size_t s = 1; s < shape.size(); s++)
                     if (ov::DimensionTracker::get_label(shape[s]))
-                        IE_THROW(NotImplemented)
-                            << "Auto-batching operates only networks with outputs batched by 0th dimension";
+                        OPENVINO_THROW("Auto-batching operates only networks with outputs batched by 0th dimension");
             }
         }
         if (!batched_inputs.size() || !batched_outputs.size())
-            IE_THROW(NotImplemented)
-                << "Auto-batching supports only networks with inputs/outputs featuring batched dim!";
-    } catch (const InferenceEngine::Exception&) {
-        metaDevice.batch_for_device = 1;
+            OPENVINO_THROW("Auto-batching supports only networks with inputs/outputs featuring batched dim!");
+    } catch (const ov::Exception&) {
+        meta_device.device_batch_size = 1;
     }
 
-    if (!metaDevice.batch_for_device) {
-        unsigned int requests = 0;
+    if (!meta_device.device_batch_size) {
         // batch size is not set explicitly via device name e.g. BATCH:GPU(4)
         // let's query the optimal batch size
-        std::map<std::string, InferenceEngine::Parameter> options;
-        options["MODEL_PTR"] = std::const_pointer_cast<ngraph::Function>(network.getFunction());
-        auto optBatchSize = core->GetMetric(deviceName, METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
-        auto res = core->GetConfig(deviceName, CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS)).as<std::string>();
-        requests = InferenceEngine::PerfHintsConfig::CheckPerformanceHintRequestValue(res);
-        const auto& reqs = user_config.find(CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS));
-        if (reqs != user_config.end())
-            requests = static_cast<unsigned int>(
-                InferenceEngine::PerfHintsConfig::CheckPerformanceHintRequestValue(reqs->second));
+        ov::AnyMap options;
+        options["MODEL_PTR"] = std::const_pointer_cast<ngraph::Function>(model);
+        auto opt_batch_size =
+            core->get_property(device_name, ov::optimal_batch_size.name(), options).as<unsigned int>();
```

Comment:
is it possible to write so?
```suggestion
 unsigned int opt_batch_size = core->get_property(device_name, ov::optimal_batch_size, options);
```